catalog/transformer/catalog-transformer-html/src/test/java/ddf/catalog/transformer/html/HtmlMetacardUtilityTest.java

```diff
@@ -0,0 +1,215 @@
+    List<HtmlMetacardModel> metacardModelList = new ArrayList<>();
+    metacardModelList.add(new HtmlMetacardModel(metacard, categories));
+
+    Document doc = Jsoup.parse(htmlMetacardUtility.buildHtml(metacardModelList));
+
+    assertThat(doc.select(METACARD_CLASS), hasSize(metacardModelList.size()));
+    assertThat(doc.select(CATEGORY_CLASS), hasSize(categories.size()));
+  }
+
+  @Test
+  public void testAssociationsAttributes() {
+    MetacardImpl metacard = new MetacardImpl();
+    metacard.setAttribute(Associations.RELATED, "");
+    metacard.setAttribute(Associations.DERIVED, "");
+    metacard.setAttribute(Associations.EXTERNAL, "");
+
+    List<HtmlExportCategory> categories = new ArrayList<>();
+    categories.add(getHtmlCategoryModel(metacard, "Associations", associationsList));
+
+    List<HtmlMetacardModel> metacardModelList = new ArrayList<>();
+    metacardModelList.add(new HtmlMetacardModel(metacard, categories));
+
+    Document doc = Jsoup.parse(htmlMetacardUtility.buildHtml(metacardModelList));
+
+    assertThat(doc.select(METACARD_CLASS), hasSize(metacardModelList.size()));
+    assertThat(doc.select(CATEGORY_CLASS), hasSize(categories.size()));
+    assertThat(doc.select(CATEGORY_TABLE_CLASS), hasSize(categories.size()));
+    assertThat(doc.select(METACARD_ATTRIBUTE_CLASS), hasSize(associationsList.size()));
+  }
+
+  @Test
+  public void testEmptyAttributeValue() {
+    MetacardImpl metacard = new MetacardImpl();
+    metacard.setAttribute(Associations.RELATED, null);
+    metacard.setAttribute(Associations.DERIVED, "");
+    metacard.setAttribute(Associations.EXTERNAL, "");
+
+    List<HtmlExportCategory> categories = new ArrayList<>();
+    categories.add(getHtmlCategoryModel(metacard, "Associations", associationsList));
+
+    List<HtmlMetacardModel> metacardModelList = new ArrayList<>();
+    metacardModelList.add(new HtmlMetacardModel(metacard, categories));
+
+    Document doc = Jsoup.parse(htmlMetacardUtility.buildHtml(metacardModelList));
+
+    assertThat(doc.select(METACARD_ATTRIBUTE_CLASS), hasSize(associationsList.size()));
+    assertThat(doc.select(EMPTY_ATTRIBUTE_CLASS), hasSize(1));
+  }
+
+  @Test
+  public void testMediaAttributeValue() {
+    MetacardImpl metacard = new MetacardImpl();
+    metacard.setThumbnail(new byte[] {});
+
+    List<HtmlExportCategory> categories = new ArrayList<>();
+    categories.add(getHtmlCategoryModel(metacard, "Core", coreList));
+
+    List<HtmlMetacardModel> metacardModelList = new ArrayList<>();
+    metacardModelList.add(new HtmlMetacardModel(metacard, categories));
+
+    Document doc = Jsoup.parse(htmlMetacardUtility.buildHtml(metacardModelList));
+
+    assertThat(doc.select(MEDIA_ATTRIBUTE_CLASS), hasSize(1));
+  }
+
+  @Test
+  public void testNullHtmlMetacardModelList() {
+    assertNull(this.htmlMetacardUtility.buildHtml(null));
+  }
+
+  @Test(expected = NullPointerException.class)
```

Comment:
We should try to prevent an NPE and try to throw some other type of exception

Reply:
```
    if (model == null || template == null) {
      return null;
    }